WARNINGS.md

```diff
@@ -796,6 +797,45 @@ may never see the warning.
 
 --------------------------------------------------------------------------------
 
+## <a name="provider-params"></a>Calls to 'provider' should specify a list of fields and a documentation
+
+  * Category name: `provider-params`
+  * Automatic fix: no
+
+Calls to `provider` should specify a documentation string and a list of fields:
+
+```python
+fooInfo = provider(
```

Comment:
Let's give a better example, for instance:
```
server_address = provider(
   "The address of an HTTP server. Fields are host (string) and port (int)."
  fields = ["host", "port"])
```

Reply:
Thanks, updated.
I've updated the name to ServerAddressInfo, because that's the convention (CamlCase as it's considered as a type).